examples/SatisfiesAnyOf.purs

```diff
@@ -0,0 +1,111 @@
+-- | This module creates a transaction that tries to pay
+-- | 2 Ada to the `AlwaysSucceeds` script address by
+-- | providing an incorrect DatumHash in the first constraints
+-- | list in `mustSatisfyAnyOf`. The transaction then
+-- | goes through with the second constraints list.
+-- | In a second transaction, the funds are returned to the wallet.
+module Ctl.Examples.SatisfiesAnyOf
+  ( example
+  , main
+  , payToSatisfiesAnyOf
+  , spendFromSatisfiesAnyOf
+  ) where
+
+import Contract.Prelude
+
+import Contract.Address (scriptHashAddress)
+import Contract.Config (ConfigParams, testnetNamiConfig)
+import Contract.Log (logInfo')
+import Contract.Monad (Contract, launchAff_, runContract)
+import Contract.PlutusData
+  ( Datum(Datum)
+  , PlutusData(Integer)
+  , unitDatum
+  , unitRedeemer
+  )
+import Contract.ScriptLookups as Lookups
+import Contract.Scripts (Validator, ValidatorHash, validatorHash)
+import Contract.Test.E2E (publishTestFeedback)
+import Contract.Transaction (TransactionHash, awaitTxConfirmed, lookupTxHash)
+import Contract.TxConstraints (TxConstraints)
+import Contract.TxConstraints as Constraints
+import Contract.Utxos (utxosAt)
+import Contract.Value as Value
+import Control.Monad.Error.Class (liftMaybe)
+import Ctl.Examples.AlwaysSucceeds (alwaysSucceedsScript) as AlwaysSucceeds
+import Ctl.Examples.Helpers (buildBalanceSignAndSubmitTx) as Helpers
+import Ctl.Internal.Hashing (datumHash) as Hashing
+import Ctl.Internal.Plutus.Types.TransactionUnspentOutput (_input)
+import Data.Array (head)
+import Data.BigInt as BigInt
+import Data.Lens (view)
+import Data.Map as Map
+import Effect.Exception (error)
+
+main :: Effect Unit
+main = example testnetNamiConfig
+
+example :: ConfigParams () -> Effect Unit
+example cfg = launchAff_ do
+  runContract cfg do
+    logInfo' "Running Examples.SatisfiesAnyOf"
+    validator <- AlwaysSucceeds.alwaysSucceedsScript
```

Comment:
There's no need for this script at all. Remove everything except the use of the constraint. It's not even needed to send the tx.

Reply:
but I need some `ValidatorHash` for the `mustPayToScript` function. Without the `mustPayToScript` constraint I get an Ogmios error because there is no Datum in the transaction.

Reply:
Just don't send the transaction.